libs/partners/zenguard/langchain_zenguard/tools.py

```diff
@@ -0,0 +1,87 @@
+import os
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+import requests
+from langchain_core.pydantic_v1 import BaseModel, Field, ValidationError, validator
+from langchain_core.tools import BaseTool
+
+
+class Detector(str, Enum):
+    ALLOWED_TOPICS = "allowed_subjects"
+    BANNED_TOPICS = "banned_subjects"
+    PROMPT_INJECTION = "prompt_injection"
+    KEYWORDS = "keywords"
+    PII = "pii"
+    SECRETS = "secrets"
+    TOXICITY = "toxicity"
+
+class DetectorAPI(str, Enum):
+    ALLOWED_TOPICS = "v1/detect/topics/allowed"
+    BANNED_TOPICS = "v1/detect/topics/banned"
+    PROMPT_INJECTION = "v1/detect/prompt_injection"
+    KEYWORDS = "v1/detect/keywords"
+    PII = "v1/detect/pii"
+    SECRETS = "v1/detect/secrets"
+    TOXICITY = "v1/detect/toxicity"
+
+class ZenGuardInput(BaseModel):
+    prompts: List[str] = Field(..., min_items=1, min_length=1)
+    detectors: List[Detector] = Field(..., min_items=1)
+    in_parallel: bool = Field(default=True)
+
+class ZenGuardTool(BaseTool):
+    name = "ZenGuard"
+    description = "Fastest LLM Security Guardrails"
+    args_schema = ZenGuardInput
+    return_direct = True
```

Comment:
what does this do?

Reply:
return direct need for Agents with using tools, Setting this to True means

that after the tool is called, the AgentExecutor will stop looping. AgentExecutor this Agent that is using tools.

I thought this field was needed to return output data directly when calling the run, invoke, batch methods.)